### PR TITLE
SendFile

### PR DESCRIPTION
**Pull request by Bogdan-D**
# Submit a pull request

## CLA

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [x] The code changes follow best practices
- [x] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -138,6 +146,73 @@ func (c *Client) VerifyWebhook(body []byte, signature []byte) (valid bool) {
...
+
+	return form.Writer.CreatePart(h)
+}
+
+func (c *Client) sendFile(url string, options SendFileRequest) (string, error) {
+	if options.User == nil {
+		return "", errors.New("user is nil")
+	}
+
+	body := new(bytes.Buffer)
```
Comment (pascaldekloe) [inline client.go]: Entire file in memory? 😏

File: client.go [hunk 2]
```diff
@@ -138,6 +146,73 @@ func (c *Client) VerifyWebhook(body []byte, signature []byte) (valid bool) {
...
+	form := multipartForm{multipart.NewWriter(body)}
+	file, err := form.CreateFormFile("file", options.FileName, options.ContentType)
+	if err != nil {
+		return "", err
+	}
+	_, err = io.Copy(file, options.Reader)
+	if err != nil {
+		return "", err
+	}
+	user, err := form.CreateFormField("user")
```
[the remark this answers is not in the document]
Comment (Bogdan-D) [reply client.go]: chat backend require user to be set also within file data

File: client.go (hunk 1 above)
Comment (Bogdan-D) [reply client.go]: fixed to tmpfile

File: client.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Bogdan-D) [reply client.go]: got what you mean, +1

**Pull request merged**